plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java

```diff
@@ -221,68 +238,76 @@ public IcebergStatistics build()
                 io.trino.spi.type.Type type,
-                Object lowerBound,
-                Object upperBound,
+                @Nullable Object lowerBound,
+                @Nullable Object upperBound,
                 Optional<Long> nullCount,
                 long recordCount)
         {
             // If this column is only nulls for this file, don't update or invalidate min/max statistics
-            if (nullCount.isEmpty() || nullCount.get() != recordCount) {
-                columnStatistics.putIfAbsent(id, new ColumnStatistics(type, typeManager, lowerBound, upperBound));
-                columnStatistics.get(id).updateMinMax(lowerBound, upperBound);
+            MethodHandle comparisonHandle = typeToComparisonHandle.get(type);
+            if (comparisonHandle != null && (nullCount.isEmpty() || nullCount.get() != recordCount)) {
+                columnStatistics.computeIfAbsent(id, ignored -> new ColumnStatistics(lowerBound, upperBound, comparisonHandle))
+                        .updateMinMax(lowerBound, upperBound);
             }
         }
 
-        private static Map<Integer, Object> convertBounds(Map<Integer, Type.PrimitiveType> idToTypeMapping, Map<Integer, ByteBuffer> idToMetricMap)
+        /**
+         * Converts a file's column bounds to a Map from field id to Iceberg Object representation
+         * @param idToMetricMap A Map from field id to Iceberg ByteBuffer representation
+         * @return A Map from field id to Iceberg Object representation
+         */
+        private Optional<Map<Integer, Object>> convertBounds(@Nullable Map<Integer, ByteBuffer> idToMetricMap)
         {
             if (idToMetricMap == null) {
-                return null;
+                return Optional.empty();
             }
             ImmutableMap.Builder<Integer, Object> map = ImmutableMap.builder();
             idToMetricMap.forEach((id, value) -> {
                 Type.PrimitiveType type = idToTypeMapping.get(id);
                 verify(type != null, "No type for column id %s, known types: %s", id, idToTypeMapping);
-                map.put(id, Conversions.fromByteBuffer(type, value));
+                Object icebergRepresentation = Conversions.fromByteBuffer(type, value);
+                if (icebergRepresentation != null) {
+                    map.put(id, icebergRepresentation);
+                }
             });
-            return map.build();
+            return Optional.of(map.build());
         }
     }
 
     private static class ColumnStatistics
     {
-        private final io.trino.spi.type.Type type;
-        private final TypeManager typeManager;
+        private final MethodHandle comparisonHandle;
+
         private Optional<Object> min;
         private Optional<Object> max;
 
-        public ColumnStatistics(io.trino.spi.type.Type type, TypeManager typeManager, Object initialMin, Object initialMax)
+        public ColumnStatistics(Object initialMin, Object initialMax, MethodHandle comparisonHandle)
```

Comment:
`comparisonHandle` is immutable state, so fits better as first arg (as you put the field order)